.github/workflows/release-binaries.yml

```diff
@@ -0,0 +1,49 @@
+name: Release Java Generator CLI binaries
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * *' # Every day at 2am
+
+jobs:
+  build-native-binary:
+    # Cheap way to prevent accidental releases
+    # Modify the list to add users with release permissions
+    if: contains('["manusa","rohanKanojia"]', github.actor)
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        gu-binary: [gu, gu.cmd]
+        exclude:
+          - os: ubuntu-latest
+            gu-binary: gu.cmd
+          - os: macos-latest
+            gu-binary: gu.cmd
+          - os: windows-latest
+            gu-binary: gu
+    continue-on-error: true
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+      - uses: graalvm/setup-graalvm@v1
+        with:
+          version: '22.1.0'
+          java-version: '11'
+          components: 'native-image'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install native-image component
+        run: |
+          ${{ matrix.gu-binary }} install native-image
+
+      - name: Compile native binaries
+        run: |
+          ./mvnw -pl :java-generator-cli -am clean package -Pnative -DskipTests
+
+      - name: Upload binary
```

Comment:
Add jreleaser?

https://jreleaser.org/guide/latest/index.html

Reply:
Hi! I'm the creator of JReleaser 🚀  
kcctl is perhaps the 1st ever Quarkus native application that adopted JReleaser https://github.com/kcctl/kcctl
What Gunnar and I learned when setting kcttl's build was later distilled as a Quarkus guide https://quarkus.io/guides/jreleaser
Happy to help with any questions you may have 😄

Reply:
Publishing to Maven Central works for JARs that should be consumed as dependencies but it's a bit cumbersome for binary distributions. I'm aware that MC (and Nexus) accepts zip files but the % of developers using MC as a package manager for CLIs (both as producers and consumers) is low. Package managers such as Homebrew, Sdkman, Chocolatey are far more popular for handling CLI tools IMHO.

FWIW you don't have to package the CLI as a native application, you can keep in in 100% Java. You also have the option to bundle a custom Java Runtime with jlink (your code does not need to be modular) to let consumers skip installing a Java runtime, or don't bundle a JR at all.

https://jreleaser.org/guide/latest/examples/cross-platform-jlink.html

JReleaser itself uses the fact that jlink can assemble as many target JRs (different platforms) on a single platform for its own CLI distributions.

Reply:
Right. If a jbang catalog were to be added to the fabric8io organization one could invoke the CLI with 

```
jbang fabric81o@java-generator-cli
```

The catalog can be created manually an be updated on every release, or you can also automate that with JReleaser https://jreleaser.org/guide/latest/configuration/packagers/jbang.html

Reply:
I see, so it's an executable uberjar distribution? JReleaser supports that one as well 😏 
I understand your constraints, just wanted to explain a bit more the capabilities offered by the tool and the host of options, which obviously don't have to be used all at once.

Might be worth reaching out to other RedHatters that have used JReleaser in the past with different settings. [Listing explicit names instead of accounts to avoid dragging them directly]: Max Andersen, Gunnar Morling, Guillaume Smet.

Reply:
I know JReleaser (I'm probably one of the earliest watchers of the project ;), for some reason I'm not listed in the watchers list (maybe due custom notification settings))

This PR is just considering one small module of many (~150). We definitely need to improve our **complete** release pipeline automation. JReleaser will definitely be considered for that :sunglasses:.

Reply:
Despite having submitted this PR, I think that @manusa 's arguments are pretty solid and this project is already complex enough without having to deal with these details.

The only use-case(I can think of) for running this CLI without a JVM installed is using it through Docker, as it's suggested in the [other kubernetes-client](https://github.com/kubernetes-client/java/blob/master/docs/generate-model-from-third-party-resources.md#2-run-code-generator-image) and have been requested [here](https://github.com/fabric8io/kubernetes-client/issues/3820) by @OneCricketeer .
I, honestly, see little added value in this last use-case as well, as it's extremely easy to re-use already available images to run the tool e.g.:
```bash
docker run --rm -it jbangdev/jbang-action io.fabric8:java-generator-cli:6.0.0-RC1
```

Reply:
Running from Docker is a requirement (not a feature) in the other client because generation requires Kind and other stuff. They have however an interesting approach of running the generator through a GitHub action from your own fork. This might be worth exploring.

Reply:
What's the use case for the GH Action?
Looking at the [workflow](https://github.com/kubernetes-client/java/blob/master/.github/workflows/generate-crd.yml) the very same can be easily achieved using the helpers from `jbang`, should we document it?

Reply:
In their case, since they **need** to run their "generator" in Docker, and very likely some of their users don't have a Docker daemon available (or enough resources, or whatever) it's quite convenient to have an action that runs the generation on a GitHub VM for free. It also improves the UX, since instead of requiring users to write a nasty command, they just need to fill in the action's inputs.

In our case, it would just help out for that precise case where users don't have a JVM available (or JBang). And just because (feature parity).